nixos/modules/installer/cd-dvd/sd-image.nix

```diff
@@ -16,7 +12,7 @@
 with lib;
 
 let
-  rootfsImage = pkgs.callPackage ../../../lib/make-ext4-fs.nix ({
+  rootfsImage = pkgs.callPackage ../../../lib/make-f2fs-fs.nix ({
```

Comment:
This PR is still changing the default FS without giving an option to choose the FS.

This is a blocker from my point of view. I would vote to keep keep the default `ext4` and add an option to choose f2fs.

Moreover, the PR contains mulriple other changes. Here are the other changes in this PR I was able to identify and believe can be split out:
- renaming `firmwareXYZ` options to `bootXYZ`
- changing the `populateRootCommands` option
- changing the `/boot/firmware` filesystem to `/boot`
- moving the auto-expand option to the `stage-1-init.sh` script (and adding the `autoExpand` option to the `filesystems` module)

Could you please create a PR where the only change is making the root FS type configurable? The other changes should go into individual PRs.